clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1053,6 +1053,7 @@ public void assign(Collection<TopicPartition> partitions) {
      *             function is called
      * @throws org.apache.kafka.common.errors.InterruptException if the calling thread is interrupted before or while
      *             this function is called
+     * @throws org.apache.kafka.common.errors.AuthenticationFailedException if caller's authentication fails
```

Comment:
Is this type correct? It seems like we're only sure that it is an instance of `AuthenticationException`. Also, is `poll()` the only API that can receive this? It seems like it should be possible from any of the blocking APIs.

Reply:
A few methods are still missing `AuthenticationException` (e.g. `partitionsFor` in producer and consumer)